File: pset2/vigenere/vigenere.c
// Program that encrypts messages using Vigenère’s cipher
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

//argc is argument count int type variable - how much stuff did the user type?
//argv is an array of strings = user input
int main(int argc, string argv[])
{
    int j;
    int k;  // to hold current character of the alphabet
    int count = 0;  // Each time we pass through the loop - use the next character in the keyword

    // Make sure that user enters only 1 secret key
    if (argc != 2)
    {
        printf("Usage: ./vigenere k\nPlease enter only 1 key:\n");
        return 1;   // signals error - try again
    }

    // Variable for secret keyword from command-line
    string key = argv[1];
    // Check that only alphabetic characters were entered as input
    for (int i = 0; i < strlen(key); i++)
    {
        if (isalpha(key[i]) == false)
        {
            printf("Keyword may only contain alphabetic characters.\n");
            //  Error and exit
            return 1;
        }
    }

    // User input for word to cipher
    string plainText = get_string("plaintext:");

    //  Print the word ciphertext to user
    printf("ciphertext: ");

    //  Loop through plaintext characters
    for (int i = 0; i < strlen(plainText); i++)
    {
        //  If char is lowercase = print lowercase cipher
        if (islower(plainText[i]))
        {
            j = (count % strlen(key));  // number of loop % length of keyword
            k = (tolower(key[j]) - 97);  // holding current alpha character of key
            count += 1; // increase loop by 1
            printf("%c", (((plainText[i] + k) - 97) % 26) + 97);
        }

        else if (isupper(plainText[i]))
        {
            j = (count % strlen(key)); // number of loop % length of keyword
            k = (tolower(key[j]) - 97);  // holding current alpha character of key
            count += 1; // increase loop by 1
            printf("%c", (((plainText[i] + k) - 65) % 26) + 65);
        }

        // Print all chars OTHER than alphabetic
        else
        {
            //If the char is not alpha (punctuation etc.) -then print its char
            printf("%c", plainText[i]);
        }
    }

    // Print a new line
    printf("\n");

    // Return 0 and exit program
    return 0;
}

File: pset6/vigenere/vigenere.py
# Program that encrypts messages using Vigenère’s cipher
from cs50 import get_string
from sys import argv
import string


def main():

    if len(argv) != 2:
        print("Usage: python vigenere.py key")
        exit(1)

    if argv[1].isalpha() == False:
        print("Key may only contain alphabetic characters.")
        exit(1)

    # prompt the user for a string of plaintext, p
    plain_text = get_string("plaintext: ")

    print("ciphertext: ", end="")

    key = argv[1]
    key = list(key)
    plain_text = list(plain_text)

    # loop through the key
    j = 0

    # for i in range(len(p)):
    for c in plain_text:

        # If char is not alpha - just print it
        if not c.isalpha():
            print(c, end="")
            continue

         # If char is uppercase, offset ascii by 65
        if c.isupper():

            # to get int index between 0-25 of plaintext character
            p = ord(c) - 65

            # to hold current alpha character of key as an integer
            k = (ord(key[j % len(key)].upper()) - 65)

            # get cipher's uppercase letter ascii integer
            # c = (((ord(p[i]) - ord('A')) + k) % 26) + 65
            c = (p + k) % 26

            # increment loop by 1
            j += 1
            if j == len(key):
                j = 0

             # print character
            print(chr(c + 65), end="")
            continue

         # If char is uppercase, offset ascii by 65
        if c.islower():

            # to get int index between 0-25 of plaintext character
            p = ord(c) - 97

            # to get int index between 0-25 of key character -> in uppercase
            k = (ord(key[j % len(key)].lower()) - 97)

            # get cipher's uppercase letter ascii integer
            # c = (((ord(p[i]) - ord('A')) + k) % 26) + 65
            c = (p + k) % 26

            # increment loop by 1
            j += 1
            if j == len(key):
                j = 0

            # print character -> need to add 97 to get it in the lowercase
            print(chr(c + 97), end="")

    # print the result (ciphertext: and ending with a newline) and exit
    print()


if __name__ == "__main__":
    main()